File: easy/ts/majority_element.ts
function majorityElement(nums: number[]): number {
  let major = 0;
  let count = 0;

  for (const num of nums) {
    if (count === 0) {
      major = num;
    };
    count += (major === num) ? 1 : -1
  }

  return major;
};

File: easy/ruby/majority_element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
  major = 0
  count = 0
  nums.each do |num|
    major = num if count.zero?
    if num == major
      count += 1
    else
      count -= 1
    end
  end

  major
end
